unified_executor/include/execution/shape.hpp

```diff
@@ -0,0 +1,35 @@
+//
+// Created by shrijit on 6/4/20.
+//
+
+#include <array>
+
+template <unsigned _size = 1, unsigned... _sizes> struct shape_t {
```

Comment:
After looking at the runtime version of the existence of `_shape` makes more sense. It should be renamed to `_dim` for better semantics.

Reply:
Even the shape wrapper seems like a good idea at this point if it works out. Though I am not sure that if the added layer would create problems somewhere else.
We can try both designs and then keep the one everyone likes.

Reply:
> I am not sure that if the added layer would create problems somewhere else

`_dim` would be a detail purely for exposition. I'd not worry about it affecting the usage.